plugins/org.eclipse.n4js.model/model/N4JS.xcore

```diff
@@ -2158,25 +2160,47 @@ abstract class N4ClassDefinition extends N4ClassifierDefinition, ThisTarget {
 	}
 }
 
+abstract class N4AbstractNamespaceDeclaration extends VariableEnvironmentElement, NamedElement {
 
-class N4NamespaceDeclaration extends N4TypeDefinition, VariableEnvironmentElement, ModifiableElement, ExportableElement, NamedElement {
 	String name
 	contains NamespaceElement[] ownedElementsRaw
-	
+}
+
+/**
+ * This type of declaration is created only from .d.ts files and represents a module declaration such as
```

Comment:
Did you consider to interpret a module declaration as a new "virtual" resource in the sense that a new resource is created while parsing that contains the ast of that module declaration? I could imagine that this would be possible and an N4JS module declaration would be obsolete. The only thing I wonder is how the URI of this virutual resource would look like.

Reply:
Yes, actually I started with something in that direction (not an entirely separate resource, but using "ordinary" `TModule`s instead of introducing `TDeclaredModule`). But at some point I stopped following this approach because I felt that it leads to too many special cases / complexity in other areas (and in areas that are more tricky to understand). Also, my gut feeling is that since we have to deal with declared modules anyway in later stages, it is better to have them represented explicitly as a concept.

However, the approach of a "virtual" resource is definitely a good idea, but it is difficult to tell what leads to the simplest solution without investigating this further. Therefore, we should definitely keep it in mind and reconsider it.

As discussed/decided in meeting: let's merge as is for now and come back to it later.